hoodie-client/src/main/java/com/uber/hoodie/index/hbase/HBaseIndex.java

```diff
@@ -303,14 +337,133 @@ private void doPutsAndDeletes(HTable hTable, List<Put> puts, List<Delete> delete
     hTable.flushCommits();
     puts.clear();
     deletes.clear();
+    sleepForTime(SLEEP_TIME_MILLISECONDS);
+  }
+
+  private static void sleepForTime(int sleepTimeMs) {
+    try {
+      Thread.sleep(sleepTimeMs);
+    } catch (InterruptedException e) {
+      logger.error(e);
+      throw new RuntimeException(e);
+    }
   }
 
   @Override
   public JavaRDD<WriteStatus> updateLocation(JavaRDD<WriteStatus> writeStatusRDD, JavaSparkContext jsc,
       HoodieTable<T> hoodieTable) {
+    setPutBatchSize(writeStatusRDD, jsc);
     return writeStatusRDD.mapPartitionsWithIndex(updateLocationFunction(), true);
   }
 
+  private void setPutBatchSize(JavaRDD<WriteStatus> writeStatusRDD,
+      final JavaSparkContext jsc) {
+    if (config.getHbaseIndexPutBatchSizeAutoCompute()) {
+      SparkConf conf = jsc.getConf();
+      int maxExecutors = conf.getInt(DEFAULT_SPARK_EXECUTOR_INSTANCES_CONFIG_NAME, 1);
```

Comment:
Move 1 to a constant and use in both places.

Reply:
From what I have noticed in open-source code, not every literal/defaults are extracted into constants unless it is required publicly or in multiple places. I would prefer leaving these contained into this method and not leaking it outside it as it does not make sense to set these to anything else and are not re-used anywhere too. Let me know if you still prefer it to be extracted to constant.

Reply:
My concern is it's very easily ignored and users may not understand the default behavior, but I share your concern and I think that makes sense.